src/dialects/oracle/query.js

```diff
@@ -0,0 +1,672 @@
+// Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved
+
+'use strict';
+
+const AbstractQuery = require('../abstract/query');
+const SequelizeErrors = require('../../errors');
+const parserStore = require('../parserStore')('oracle');
+const _ = require('lodash');
+const Utils = require('../../utils');
+const { logger } = require('../../utils/logger');
+
+const debug = logger.debugContext('sql:oracle');
+
+class Query extends AbstractQuery {
```

Comment:
Partially at @ephys do we want to name this `OracleQuery`?

Reply:
I missed it but yes :)

Reply:
Used OracleQuery